packages/core/src/external/aws/s3.ts

```diff
@@ -98,8 +98,14 @@ export class S3Utils {
     key: string,
     bucket: string
   ): Promise<
-    | { exists: true; size: number; contentType: string; eTag?: string }
-    | { exists: false; size?: never; contentType?: never; eTag?: never }
+    | {
+        exists: true;
+        size: number;
+        contentType: string;
+        eTag?: string;
+        dateCreated: Date | undefined;
```

Comment:
Any particular reason not to keep the pattern of `createdAt`?

Reply:
no reason. Will change it for consistency